openwisp_network_topology/api/views.py

```diff
@@ -28,8 +72,24 @@ class NetworkCollectionView(generics.ListAPIView):
     serializer_class = NetworkGraphSerializer
     queryset = Topology.objects.filter(published=True)
 
-
-class NetworkGraphView(generics.RetrieveAPIView):
+    def get_queryset(self):
+        queryset = super().get_queryset()
+        user = self.request.user
+        if user.is_superuser:
+            return queryset
+        if user.is_authenticated:
+            queryset = queryset.filter(organization__in=user.organizations_managed)
+        pks = []
+        for obj in queryset:
+            try:
+                self.check_object_permissions(self.request, obj)
```

Comment:
From [ListModelMixins](https://github.com/encode/django-rest-framework/blob/master/rest_framework/mixins.py#L37) we see that the ListAPIView doesn't check for permissions on objects. so we have to explicitly check for each object

Reply:
Again, this is not necessary Noumbissi, there's no object level permission implementation on OpenWISP, we check two things: the permission from `django.contrib.auth.models.Permission` (which we are already doing) and the organization membership with `is_admin=True`. This check is not necessary and generates unnecessary queries, please remove it.